typings/index.d.ts

```diff
@@ -14,11 +14,85 @@ declare module 'discord.js' {
   import { ChildProcess } from 'child_process';
   import { EventEmitter } from 'events';
   import { PathLike } from 'fs';
+  import { Response } from 'node-fetch';
   import { Readable, Stream, Writable } from 'stream';
+  import { URLSearchParams } from 'url';
   import * as WebSocket from 'ws';
 
   export const version: string;
 
+  //#region Rest
+
+  class AsyncQueue {
+    // tslint:disable-next-line:ban-types
+    private readonly promises: { promise: Promise<void>; resolve: Function }[];
```

Comment:
```suggestion
    private readonly promises: { promise: Promise<void>; resolve: (value?: any) => void }[];
```

Reply:
From the original TypeScript code, there's this interface:

```typescript
interface InternalAsyncQueueDeferredPromise {
	resolve(): void;
	promise: Promise<void>;
}
```

And `AsyncQueue#promises` is `InternalAsyncQueueDeferredPromise[]`. Therefore:

```suggestion
    private readonly promises: { promise: Promise<void>; resolve(): void }[];
```